### PR TITLE
Fixed - Fatal Error when save configurable product in Magento 2.2.5 #18082

### PR DESCRIPTION
**Pull request by Shubham0492**
**Description**
I have checked that if there is any validation error exist then after click save button configurable-matrix data remove.You can check the file app/code/Magento/ConfigurableProduct/view/adminhtml/web/js/variations/variations.js in function serializeData due to fatal error comes.Now I have added condition that if there is any validation error exist then configurable-matrix data will not remove and after submit form all value will exist and product will successfully save.

**Fixed Issues (if relevant)**
1 - magento/magento2#18082: Fatal Error when save configurable product in Magento 2.2.5

### Manual testing scenarios
1. Add new configurable product with name and sku of 63 characters length For Example: "Amzer 203877 Kristal FULL HD Tempered Glass for OnePlus 6 Clear".
2. Now click on "Create Configuration" button.
3. Now, When you try to save it's configuration, It shows client side validation error for child products SKUs. Decrease length of child product's SKUs and then save the product. It generates a error report as attached.But after apply changes product successfully saved.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=18474) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=18474) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: Shubham-Webkul<br/>:x: shubham sharma<hr/>**shubham sharma** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

File: app/code/Magento/ConfigurableProduct/view/adminhtml/web/js/variations/variations.js [hunk 1]
```diff
@@ -406,17 +406,20 @@ define([
...
-
-            this.source.data['associated_product_ids_serialized'] =
-                JSON.stringify(this.source.data['associated_product_ids']);
-
-            delete this.source.data['associated_product_ids'];
-        },
+         serializeData: function () {
+             this.source.data['configurable-matrix-serialized'] =
+                 JSON.stringify(this.source.data['configurable-matrix']);
+             if ($('.admin__field-error').length===0) {
```
Comment (VladimirZaets) [inline app/code/Magento/ConfigurableProduct/view/adminhtml/web/js/variations/variations.js]: Hi, @ShubhamSharma967. Thanks for the collaboration. We try to not use selectors directly. You can pass the selector name as configuration property to this component.  
Feel free to ask questions or request additional information or some help.

**Comment by Shubham0492** [comment]
I have fixed direct use of selector in latest commit.  please check latest commit 8c8e9edf5432ff669614fb42f62e6af4e9d93d7c .In this commit there is no need to change following file
https://github.com/magento/magento2/blob/8c8e9edf5432ff669614fb42f62e6af4e9d93d7c/app/code/Magento/ConfigurableProduct/view/adminhtml/layout/catalog_product_attribute_edit_product_tab_variations_popup.xml
 it was change by mistake.This issue will  fix by change  in following two files : -
https://github.com/magento/magento2/blob/8c8e9edf5432ff669614fb42f62e6af4e9d93d7c/app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/super/matrix.phtml
https://github.com/magento/magento2/blob/8c8e9edf5432ff669614fb42f62e6af4e9d93d7c/app/code/Magento/ConfigurableProduct/view/adminhtml/web/js/variations/variations.js

**Comment by Shubham0492** [comment]
fixed conflict issue in this commit 34570d3dad76fa715b7141a76c9d464a065c9592

**Comment by Shubham0492** [comment]
#SQUASHTOBERFEST

**Comment by VladimirZaets** [comment]
Hi @Shubham-Webkul, currently your branch contains more than 5000 changed files, I tried to change base branch but had the same situation. Please fix your branch and create new one PR. Thanks.

**Pull request closed without merging**